src/main/java/io/bespin/java/util/Tokenizer.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Bespin: reference implementations of "big data" algorithms
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.bespin.java.util;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+
+/**
+ * Tokenizer so that we have a consistent definition of a "word".
+ */
+public class Tokenizer {
+  public static List<String> tokenize(String input) {
```

Comment:
Since the regex doesn't change, it's more efficient to compile it into a Pattern hidden into a private static variable then calling `pattern.matcher(text).replaceAll(repl)`, to avoid compiling it for every token, since this method is often in the inner loop.